### ISSUE TITLE
Travis build failing with node 10, g++ 4.8

### ISSUE THREAD

**Issue by melvincarvalho** [solid/node-solid-server#658]
Travis CI is currently passing with node 8 but failing with node 10, using g++ 4.8

Here is an example report

https://travis-ci.org/solid/node-solid-server/jobs/373706843

```
../../nan/nan.h:1969:16: note: candidate is:
In file included from ../src/deasync.cc:2:0:
/home/travis/.node-gyp/10.0.0/include/node/v8.h:3214:37: note: v8::Maybe<bool> v8::Object::SetAccessor(v8::Local<v8::Context>, v8::Local<v8::Name>, v8::AccessorNameGetterCallback, v8::AccessorNameSetterCallback, v8::MaybeLocal<v8::Value>, v8::AccessControl, v8::PropertyAttribute, v8::SideEffectType)
   V8_WARN_UNUSED_RESULT Maybe<bool> SetAccessor(
                                     ^
/home/travis/.node-gyp/10.0.0/include/node/v8.h:3214:37: note:   no known conversion for argument 2 from ‘Nan::imp::NativeGetter {aka void (*)(v8::Local<v8::Name>, const v8::PropertyCallbackInfo<v8::Value>&)}’ to ‘v8::Local<v8::Name>’
make: *** [Release/obj.target/deasync/src/deasync.o] Error 1
make: Leaving directory `/home/travis/build/solid/node-solid-server/node_modules/deasync/build'
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
```

It appears to be failing on compiling the [deasync](https://github.com/abbr/deasync) module

**Comment by melvincarvalho** [comment]
[Node 10](https://github.com/nodejs/node/blob/master/doc/changelogs/CHANGELOG_V10.md#2018-04-24-version-1000-current-jasnell) was released on 24 April and has made it into travis.  New JS Features : 

- Promise#finally
- Async generators
- Async iterators
- String#trimStart/trimEnd
- Standardized `global` object (`window` alternative)
- RegExp: `s` (dotAll) & `u` (unicode) flags, named captures, lookbehinds
- Optional `(e)` in `try {} catch(e) {}`

Currently the sleep function which depends on deasync doesnt compile with g++ 4.8

**Comment by melvincarvalho** [comment]
Errors with node 10 (aside from installing)

```
  1) ACL with WebID through X-SSL-Cert with a valid certificate should return 200:
     AssertionError: expected { Object (_readableState, readable, ...) } to have a property 'statusCode' of 200, but got 500
      at Function.assert.propertyVal (node_modules/chai/lib/chai/interface/assert.js:1001:37)
      at Context.<anonymous> (test/integration/acl-tls-test.js:1125:14)

  2) ACL with WebID through X-SSL-Cert with a valid certificate should set the User header:
     AssertionError: expected { Object (x-powered-by, vary, ...) } to have a property 'user'
      at Function.assert.propertyVal (node_modules/chai/lib/chai/interface/assert.js:1001:37)
      at Context.<anonymous> (test/integration/acl-tls-test.js:1129:14)

  3) ACL with WebID through X-SSL-Cert with an invalid certificate should return 401:
     AssertionError: expected { Object (_readableState, readable, ...) } to have a property 'statusCode' of 401, but got 500
      at Function.assert.propertyVal (node_modules/chai/lib/chai/interface/assert.js:1001:37)
      at Context.<anonymous> (test/integration/acl-tls-test.js:1164:14)

  4) HTTP COPY API should give a 404 if source document doesn't exist:
     Uncaught Error: ENOENT: no such file or directory, unlink './test/resources/accounts/localhost/sampleUser1Container/invalid-resource-copy'
      at Object.fs.unlinkSync (fs.js:1015:3)
      at cleanupFileStream (lib/ldp-copy.js:6:176)
      at WriteStream.<anonymous> (lib/ldp-copy.js:11:605)
      at onwriteError (_stream_writable.js:433:12)
      at onwrite (_stream_writable.js:469:5)
      at doWrite (_stream_writable.js:409:11)
      at writeOrBuffer (_stream_writable.js:397:5)
      at WriteStream.Writable.write (_stream_writable.js:295:11)
      at Request.ondata (internal/streams/legacy.js:15:31)
      at IncomingMessage.<anonymous> (node_modules/request/request.js:1088:12)
      at addChunk (_stream_readable.js:280:12)
      at readableAddChunk (_stream_readable.js:265:11)
      at IncomingMessage.Readable.push (_stream_readable.js:220:10)
      at HTTPParser.parserOnBody (_http_common.js:124:22)
      at TLSSocket.socketOnData (_http_client.js:421:20)
      at addChunk (_stream_readable.js:280:12)
      at readableAddChunk (_stream_readable.js:265:11)
      at TLSSocket.Readable.push (_stream_readable.js:220:10)
      at TLSWrap.onread (net.js:638:20)

  5) HTTP COPY API "after all" hook:
     Uncaught TypeError: Cannot read property 'isPending' of undefined
  



  6) "before each" hook for "should exist"
```

**Comment by melvincarvalho** [comment]
Some dates

https://github.com/nodejs/Release#release-schedule

Node 10 is brand new, wont be LTS until October.  Node 8 will be LTS until December 2019.

Suggestion is to ignore node 10 in travis, or remove it from .travis.yml for a while, until someone has time to fix this, or if someone using node 10 offers a PR.  I think it's a non trivial fix.

**Comment by RubenVerborgh** [comment]
Is already fixed in the `develop` branch: da6b7d3da23bd245b8e2192a1c624f93df781c47, 0c0b78482131c1ba32809730c029fb048488963a.
